x-pack/plugin/esql/compute/src/main/generated-src/org/elasticsearch/compute/data/DoubleArrayBlock.java

```diff
@@ -57,6 +61,16 @@ public DoubleBlock expand() {
         return new DoubleArrayBlock(values, end, firstValues, shiftNullsToExpandedPositions(), MvOrdering.UNORDERED);
     }
 
+    public static long ramBytesEstimated(double[] values, int[] firstValueIndexes, BitSet nullsMask) {
```

Comment:
Is it useful that this is `public static`? I feel like it might be useful to pre-bump a breaker, but you'd want to do that before allocating the arrays.

Reply:
Yeah, it's still a little unclear if these need to be public or not. I felt them public for now. The idea (for a subsequent PR), since the various arrays can grow independently from each other, is to account for / circuit-break the array element sizes in the builder, then compute and add the actual difference when building the block. That way we can work off reasonable estimates until we build the actual block.